app/server/endpoint/src/main/java/io/syndesis/server/endpoint/v1/handler/integration/IntegrationOverviewHelper.java

```diff
@@ -84,48 +86,50 @@ static IntegrationOverview toCurrentIntegrationOverview(final Integration integr
                 .collect(Collectors.toList()));
         }
 
-        IntegrationDeployment deployed = null;
+        final ListResult<IntegrationDeployment> deployments = dataManager.fetchAll(IntegrationDeployment.class,
+                new IdPrefixFilter<>(id + ":"), ReverseFilter.getInstance());
+
         final List<IntegrationDeployment> activeDeployments = new ArrayList<>();
-        for (final IntegrationDeployment deployment : dataManager.fetchAll(IntegrationDeployment.class,
-            new IdPrefixFilter<>(id + ":"), ReverseFilter.getInstance())) {
+        for (final IntegrationDeployment deployment : deployments) {
             builder.addDeployment(IntegrationDeploymentOverview.of(deployment));
 
-            final IntegrationDeploymentState currentState = deployment.getCurrentState();
-            if (currentState == IntegrationDeploymentState.Published) {
-                deployed = deployment;
-                builder.deploymentVersion(deployment.getVersion());
-            }
-
-            if (currentState != IntegrationDeploymentState.Unpublished) {
-                // the bet is that any integration that the user wanted to
-                // publish
-                // will have it's status != Unpublished, the reason why we can't
-                // look at the last deployment is because users can choose to
-                // deploy
-                // previous deployments, so we bet that all the Unpublished
-                // integrations are not the ones that the user don't hold the
-                // current state
-                builder.targetState(deployment.getTargetState());
-                builder.currentState(currentState);
+            if (deployment.getCurrentState().equals(IntegrationDeploymentState.Pending) ||
```

Comment:
```suggestion
            if (deployment.getCurrentState() == IntegrationDeploymentState.Pending ||
```

Reply:
Not 100% sure about including `Pending`, I might be wrong it doesn't seem like we did that in the current version.